### ISSUE TITLE
Be able to create issues and add to issues through conversations on Slack (using the new Slack Actions)

### ISSUE THREAD

**Issue by zackyap** [integrations/slack#594]
Slack released Actions as shared in the URL below:
https://slackhq.com/turn-your-conversations-into-action-7acfc3c0d3f0

It would be great to be able to:
1. Create Issues from conversations.
2. Add conversations to an issue.

![1_xck2v6twrwtcenwzwp9x7a](https://user-images.githubusercontent.com/346771/40412203-3978bd20-5ea5-11e8-9089-97300d3e498f.jpeg)

![1_jng6ahyh1npgc0m_mtl9fw](https://user-images.githubusercontent.com/346771/40412213-3d3c1c68-5ea5-11e8-9cd2-2bc957584896.gif)

**Comment by mikerudge** [comment]
Might take a look at this... see if I can contribute towards getting this in as I think it would be great!

**Comment by mikerudge** [comment]
Any contributors got any tips on how to get all repos for the user? something like `gitHubUser.client.repos.getForUser()` 

If that is the best one to use, where is the best place to get the GitHub username from?

**Comment by winzig** [comment]
Our company is considering moving from Asana to Github. This integration (which Asana already supports) would go a long way to sealing the deal. I hope it can be integrated soon.

**Comment by winzig** [comment]
Unless it's been resolved elsewhere, it's still relevant. ;-)

**Comment by hanukunduru** [comment]
Yes would definitely be a very important feature we would love to have and use

**Comment by Waldz** [comment]
Waiting for such feature very much, because PMs cant type `/github open organization/repo`

**Comment by elliottregan** [comment]
It seems like this is currently possible: https://github.com/integrations/slack#take-action

**Comment by Waldz** [comment]
No @elliottregan, it only command line interface.
What our managers miss is clickable flow: "Create issue", "Create issue from selected text"

**Comment by djkgamc** [comment]
seriously

**Comment by webdif** [comment]
Maybe we can have a command `/github open` which opens the popin without any repository selected, we have to select it ourselves in the dropdown?

For now, we can just write `/github open owner/repo`, but it can be hard/long to type…

**Comment by winzig** [comment]
The idea is that it's not just making this clickable, but since it's acting on a specific message, is able to automatically pull text (and maybe other metadata) from the Slack message being acted upon, which also can save a bunch of typing.

**Comment by stevejwikeley** [comment]
It would be great to use this action feature but also copy the conversation though to the issue with a link back to the slack conversation - ProductBoard's slack does this really well

**Comment by fikipollo** [comment]
It would be great to have this feature! Writing "/github open mycompanyrepo/my_super_long_repo_name" anytime we want to open a new issue is not cool (and can be tedious for not advanced users). Actually, there is already an option available in slack which is "Comment on Issue or PR" that works great and event pull text from the Slack message, maybe it would be as easy as extend this tool to allow create a new Issue (now it's only possible to choose an existing one).

**Comment by nerdmed** [comment]
We also just started to use slack to create github issues. It makes sense for the workflow and i agree that typing the repo names is a little unhandy but for us still doable. as @stevejwikeley mentioned it would be extermly valuable to have a loop back to slack. 

How we could imagine it to work: 

1. Someone posts a problem inside a slack channel
2. Discussion happens inside the thread for the post
3. Developer decides that it is a real issue
4. We create an issue by using `/github open` inside the thread
5. Action window opens
6. It would have an option to bring the thread discussion to the issue, or maybe just a link inside the github issue
7. It would automatically post a message that an issue is created with the link (similar to asana)
8. **Magic**: It would automatically subscribe to the issue inside the thread and post a message into the thread as soon as the issue is closed 

What do you guys think?

**Comment by TheLastTech** [comment]
add the ability to right click a point in conversation and create issue. 

* has link to point in conversation in slack 

takes text from that and image and video and creates issue

Right click a point in conversation. Add to issue. Action window -> current issues, ordered by recently opened by this user

8. magic you say?

create a room for the issue where you can specifically reply to that issue instead of going to github.

**Comment by astout** [comment]
Well since Microsoft has since blocked all internal use of Slack, I suspect slack integrations will go silent 👎

**Comment by tcbyrd** [comment]
We use Slack pretty heavily at GitHub and rely on this integration, so we're definitely not going silent. We're in a planning mode right now and trying to prioritize all the work we need to tackle. The amount of 👍 and ❤️ on the OP is definitely a signal that helps. Nothing to announce quite yet though.

**Comment by astout** [comment]
@tcbyrd good to know! Sorry for the shade! I was making assumptions and you know what they say about those.

**Comment by Deskimo** [comment]
We'd love to be able to comment on a ticket directly from our SLACK channel (which gets ticket notifications) so that the comment is added to the github ticket.

**Comment by bradstemke** [comment]
This feature is basically the only thing stopping my boss from upgrading our Github account to a paid one so we can use it with our whole team. Hoping it gets added soon

**Comment by TheLastTech** [comment]
Just the comment part @bradstemke ?

**Comment by bradstemke** [comment]
@TheLastTech oh sorry on the commenting part, just the creation. But now that I am looking further at this, it does appear you can create one from a Slack channel with the integration. So that is great! I think I misread parts of the thread above. 

Is there a way to list out current open issues with a command in Slack?

**Comment by scepticulous** [comment]
I think there are multiple feature requests in this issue. 
In order to make triaging easier and to aggregate the interactions correctly, this issue should be focused on the UX improvements of new issues mentioned in the OP (Actions or from "marked" text).
I would like everybody who had additional feature requests like "creating comments" to create a dedicated issue for that.

**Comment by inopinatus** [comment]
Here's a UX data point: anything beyond two mouse clicks, typing a title, and hitting enter, may be an excessive burden for some users (myself included, being a lazy and impatient programmer)

**Comment by marissamarym** [comment]
Hi, if anyone's still interested in the issue creation part of this, [Instantish](https://twitter.com/instantish) lets you create issues through a slash command (`/issue`) or through a Slack Action.

We're still in early access, so if you'd like to use it, you can sign up at itsinstantish.com or [DM me on Twitter](https://twitter.com/marissamary) if you have questions.

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/1459660/68716260-add68700-0558-11ea-8b4e-e82fc70a764e.gif)

**Comment by marissamarym** [comment]
Update on the above feature if anyone's interested:

[Instantish](https://twitter.com/instantish) now also lets you **add messages to _existing_ issues** by clicking on a message.

💬  You can add up to 20 messages
📹 You can add files and videos (20MB for images, 50MB for videos)
🧵  Supports threaded messages

**Comment by raulgenially** [comment]
We would be perfect if we could open the /github open with the work flow button 
![image](https://user-images.githubusercontent.com/73470842/126490561-0fea8840-a822-4bef-b00a-a3ae3e4e90ac.png)

**Comment by stevejwikeley** [comment]
It would. Could you remove me from the repo please? I left Prolific a while
back
